python/pyspark/sql/functions.py

```diff
@@ -658,23 +658,26 @@ class UserDefinedFunction(object):
 
     .. versionadded:: 1.3
     """
-    def __init__(self, func, returnType):
+    def __init__(self, func, returnType, name=None):
         self.func = func
         self.returnType = returnType
         self._broadcast = None
-        self._judf = self._create_judf()
+        self._judf = self._create_judf(name)
 
-    def _create_judf(self):
-        f = self.func  # put it in closure `func`
-        func = lambda _, it: map(lambda x: f(*x), it)
+    def _create_judf(self, name):
+        f, returnType = self.func, self.returnType  # put them in closure `func`
+        from pyspark.serializers import CloudPickleSerializer
+        CloudPickleSerializer().dumps(returnType)
```

Comment:
What is this line doing?  It looks like we're calling it only for side-effects.  Is it necessary?